src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/dq/DirectionsQuestionnaireGenerator.java

```diff
@@ -495,17 +499,25 @@ private FurtherInformation getFurtherInformation(DQ dq, CaseData caseData) {
             .build();
     }
 
-    private RequestedCourt getRequestedCourt(DQ dq) {
+    private RequestedCourt getRequestedCourt(DQ dq, String authorisation) {
         RequestedCourt rc = dq.getRequestedCourt();
-        if (rc == null) {
+        if (null != rc && null != rc.getCaseLocation()) {
+            List<LocationRefData> courtLocations = (locationRefDataService
+                .getCourtLocationsByEpimmsId(
+                    authorisation,
+                    rc.getCaseLocation().getBaseLocation()
+                ));
             return RequestedCourt.builder()
                 .requestHearingAtSpecificCourt(YES)
+                .responseCourtName(courtLocations.isEmpty() ? null :
+                                       courtLocations.stream()
+                                           .filter(id -> id.getCourtTypeId().equals(CIVIL_COURT_TYPE_ID))
+                                           .collect(Collectors.toList()).get(0).getCourtName())
```

Comment:
can we check for potential null pointer here as after filter there might not be any record left

Reply:
I make changes and make some manual test.